contracts/Backing.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/proxy/Initializable.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./common/Scale.sol";
+import "./common/Ownable.sol";
+import { ScaleStruct } from "./common/Scale.struct.sol";
+import "./interfaces/IWETH.sol";
+import "./interfaces/IRelay.sol";
+import "./interfaces/IERC20Option.sol";
+
+contract Backing is Initializable, Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    struct BridgerInfo {
+        address target;
+        uint256 timestamp;
+    }
+
+    receive() external payable {
+        assert(msg.sender == weth);
+    }
+
+    //uint256 public registerFee = 0;
+    IRelay public relay;
+    address public weth;
+    bytes public storageKey;
+
+    mapping(address => BridgerInfo) public assets;
+    mapping(uint32 => address) public history;
+
+    event NewTokenRegisted(address indexed token, string name, string symbol, uint8 decimals);
+    event BackingLock(address indexed token, address target, uint256 amount, address receiver);
+    event VerifyProof(uint32 blocknumber);
+    event RegistCompleted(address token, address target);
+    event RedeemTokenEvent(address token, address target, address receipt, uint256 amount);
+
+    function initialize(address _relay, address _weth) public initializer {
+        ownableConstructor();
+        relay = IRelay(_relay);
+        weth = _weth;
+    }
+
+    function setStorageKey(bytes memory key) external onlyOwner {
+        storageKey = key;
+    }
+
+    function registerToken(address token) external {
+        require(assets[token].timestamp == 0, "asset has been registed");
+        assets[token] = BridgerInfo(address(0), block.timestamp);
+
+        string memory name = IERC20Option(token).name();
+        string memory symbol = IERC20Option(token).symbol();
+        uint8 decimals = IERC20Option(token).decimals();
+        emit NewTokenRegisted(
+            token,
+            name,
+            symbol,
+            decimals
+        );
+    }
+
+    function crossSendToken(address token, address recipient, uint256 amount) external {
+        require(amount > 0, "balance is zero");
+        require(assets[token].target != address(0), "asset has not been registed");
+        IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
+        emit BackingLock(token, assets[token].target, amount, recipient);
+    }
+
+    function crossSendETH(address recipient) external payable {
```

Comment:
To keep protocol simple, prefer to not directly support this.

A third party and easily setup a contract for the ETH-WETH conversion, we can omit WETH address here.

Reply:
ok, I will use periphery contract to do this work. And keep it simple.